Library/Formula/llvm.rb

```diff
@@ -198,12 +196,6 @@ def install
 
     args << "-DLLVM_ENABLE_RTTI=On" if build.with? "rtti"
 
-    if build.with? "assertions"
-      args << "-DLLVM_ENABLE_ASSERTIONS=On"
-    else
-      args << "-DCMAKE_CXX_FLAGS_RELEASE='-DNDEBUG'"
```

Comment:
At least based on the LLVM website, assertions are only enabled by default in Debug builds. CMake also includes it in the flags for Release builds, although I'm not sure if Homebrew removes that during a build...

At the very least I was able to use clang with its sanitizers for compiling/testing programs, so it seems alright, although I don't use the more complex features right now